### PR TITLE
APPS-6813: App Platform logs w/o deploy id + app.pending_deployment

### PR DESCRIPTION
**Pull request by jcodybaker**
This PR adds the pending_deployment field to the apps structure.  It also adds two new variants of the App Platform log operations.

Draft until upstream changes are merged and deployed.

### CODE REVIEW

File: specification/resources/apps/responses/list_logs_aggregate.yml [hunk 1]
```diff
@@ -1,16 +0,0 @@
-description: A JSON object with urls that point to archived logs
```
Comment (jcodybaker) [inline specification/resources/apps/responses/list_logs_aggregate.yml:1]: https://digitalocean.slack.com/archives/CASLGBDGW/p1675880510481499?thread_ts=1675875895.132299&cid=CASLGBDGW

**Review by kamaln7 — DISMISSED** [review]

File: specification/DigitalOcean-public.v2.yaml [hunk 2]
```diff
@@ -581,6 +581,10 @@ paths:
     put:
       $ref: 'resources/apps/apps_update.yml'
 
+  /v2/apps/{app_id}/components/{component_name}/logs:
+    get:
+      $ref: 'resources/apps/apps_get_logsActiveDeployment.yml'
```
Comment (danaelhe) [inline specification/DigitalOcean-public.v2.yaml:586]: this is very minor- but could you convert this to be all snakecase? 
```suggestion
      $ref: 'resources/apps/apps_get_logs_active_deployment.yml'
```

I know we have other resources that follow the snake_case_camelCase pattern, but we're trying to move away from that.

File: specification/DigitalOcean-public.v2.yaml [hunk 3]
```diff
@@ -603,6 +607,10 @@ paths:
     get:
       $ref: 'resources/apps/apps_get_logsAggregate.yml'
 
+  /v2/apps/{app_id}/logs:
+    get:
+      $ref: 'resources/apps/apps_get_logsActiveDeploymentAggregate.yml'
```
Comment (danaelhe) [inline specification/DigitalOcean-public.v2.yaml:612]:
```suggestion
      $ref: 'resources/apps/apps_get_logs_active_deployment_aggregate.yml'
```

File: specification/resources/apps/apps_get_logsActiveDeployment.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+operationId: apps_get_logsActiveDeployment
```
Comment (danaelhe) [inline specification/resources/apps/apps_get_logsActiveDeployment.yml:1]:
```suggestion
operationId: apps_get_logs_active_deployment
```

File: specification/resources/apps/apps_get_logsActiveDeployment.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    $ref: "../../shared/responses/too_many_requests.yml"
+
+  "500":
+    $ref: ../../shared/responses/server_error.yml
+
+  default:
+    $ref: ../../shared/responses/unexpected_error.yml
+
+x-codeSamples:
+  - $ref: 'examples/curl/apps_get_logsActiveDeployment.yml'
```
Comment (danaelhe) [inline specification/resources/apps/apps_get_logsActiveDeployment.yml:42]:
```suggestion
  - $ref: 'examples/curl/apps_get_logs_active_deployment.yml'
```

File: specification/resources/apps/apps_get_logsActiveDeploymentAggregate.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    $ref: "../../shared/responses/too_many_requests.yml"
+
+  "500":
+    $ref: ../../shared/responses/server_error.yml
+
+  default:
+    $ref: ../../shared/responses/unexpected_error.yml
+
+x-codeSamples:
+  - $ref: 'examples/curl/apps_get_logsActiveDeploymentAggregate.yml'
```
Comment (danaelhe) [inline specification/resources/apps/apps_get_logsActiveDeploymentAggregate.yml:41]:
```suggestion
  - $ref: 'examples/curl/apps_get_logs_active_deployment_aggregate.yml'
```

**Review by danaelhe — COMMENTED** [review]
Looks great! Just had a minor nit-pick on the naming pattern convention we'd like to follow.

File: specification/resources/apps/apps_get_logsActiveDeploymentAggregate.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
+operationId: apps_get_logsActiveDeploymentAggregate
+
+summary: Retrieve Active Deployment Aggregate Logs
+
+description: Retrieve the logs of the active deployment if one exists. The 
+  response will include links to either real-time logs of an in-progress or
+  active deployment or archived logs of a past deployment. Note log_type=BUILD
+  logs will return logs associated with the current active deployment (being
+  served). To view build logs associated with in-progress build, the query must
+  explicitly reference the deployment id.
```
Comment (danaelhe) [inline specification/resources/apps/apps_get_logsActiveDeploymentAggregate.yml:10]: This has the exact description as "Retrieve Active Deployment Logs"- what's the distinction between the two?

**Review by danaelhe — APPROVED** [review]
👍

**Pull request merged**